frontend/test/app/components_basic_test.cljs

```diff
@@ -84,127 +88,157 @@
                      new-state
                      (:parent-id shape1))
 
-                    file (dwlh/get-local-file new-state)]
+                    file (wsh/get-local-file new-state)]
 
-                (t/is (= (:name shape1) "Rect 1"))
+                (t/is (= (:name shape1) "Rect-1"))
                 (t/is (= (:name group) "Component-1"))
                 (t/is (= (:name component) "Component-1"))
-                (t/is (= (:name c-shape1) "Rect 1"))
+                (t/is (= (:name c-shape1) "Rect-1"))
                 (t/is (= (:name c-group) "Component-1"))
 
                 (thl/is-from-file group file))))
 
           (rx/subs done #(throw %))))))
 
-;; FAILING
-#_(t/deftest test-add-component-from-several-shapes
+(t/deftest test-add-component-from-several-shapes
   (t/async
    done
    (let [state (-> thp/initial-state
                    (thp/sample-page)
                    (thp/sample-shape :shape1 :rect
-                                     {:name "Rect 1"})
+                                     {:name "Rect-1"})
                    (thp/sample-shape :shape2 :rect
-                                     {:name "Rect 2"}))]
-
-     (->> state
-          (the/do-update (dw/select-shapes (lks/set
-                                            (thp/id :shape1)
-                                            (thp/id :shape2))))
-          (the/do-watch-update dwl/add-component)
-          (rx/do
-            (fn [new-state]
+                                     {:name "Rect-2"}))
+         store (the/prepare-store state done
+           (fn [new-state]
```

Comment:
This indentation looks wrong

Reply:
Solved. Compacted the commits to not have a very ugly merge.